sphinx/roles.py

```diff
@@ -322,7 +353,7 @@ def run(self) -> tuple[list[Node], list[system_message]]:
         options = self.options.copy()
         matched = self.abbr_re.search(self.text)
         if matched:
-            text = self.text[:matched.start()].strip()
+            text = self.text[: matched.start()].strip()
```

Comment:
That's a definite loss of readability IMO.

Reply:
This is PEP compliant, and has already all been discussed in black/rust: https://github.com/astral-sh/ruff/pull/8654

I honestly fail to see how this reduces readability

Reply:
Yes, but that's only if have something on the left and the right. PEP also says:

> in a slice the colon acts like a binary operator, and should have equal amounts [of spaces] on either side (treating it as the operator with the lowest priority).

So, no space is also compliant technically speaking. In addition, in most code I've seen, when you only have a single position, there is no space before or after. In my local python files I have:

```
grep -rP '\[[\S ]* :\]' --include='*.py' | wc -l
596
grep -rP '\[[\S ]*\S:\]' --include='*.py' | wc -l
3970
```

Personally, if I feel `x = a[1 :]` is less readable instead of `a[1:]`. Also, most of the files in Sphinx do not put a space after/before the colon so if we keep this rule the diff will be larger for the rest of the files.

Reply:
> if we keep this rule the diff will be larger for the rest of the files

this rule is not configurable

if you want it changed you would need to open an issue on ruff

Reply:
Wait, can't you say that you don't want the formatter to do that actually? Like, can't you tell it not to add a space? Also, we could add the 'E203' rule (which is disabled in flake8) but it needs the '--preview' flag for ruff.

Reply:
> Wait, can't you say that you don't want the formatter to do that actually? Like, can't you tell it not to add a space? 

No you can't

> Also, we could add the 'E203' rule

I think you didn't look at my link 😄 they literally changed the E203 to allow the space: https://github.com/astral-sh/ruff/pull/8654

Reply:
Ah I actually I misunderstood what they did. For me they said "our formatter does not make the change if you have E203 enabled" but actually it's not the case.

Ok, I won't say anything then since we cannot configure it.